pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -69,13 +71,15 @@
 
 public class MultiStageBrokerRequestHandler extends BaseBrokerRequestHandler {
   private static final Logger LOGGER = LoggerFactory.getLogger(MultiStageBrokerRequestHandler.class);
+  private final Random _random = new Random();
```

Comment:
had some offline discussion, can we do something like
```
requestID = ((Long)((hash(brokerId) << 32)) + _requestIdGenerator.incrementAndGet()
```
encode brokerID and requestID with high/low bits?

this way we can still masked out the lower bit to get the exact ordering of the requests when we look at the requestID. this will be better in terms of debuggability.

Reply:
Added base-2 masking. We can also do base-10 masking to make debugging even easier. Something like:

```
brokerHash * 10^9 + (requestId.incrementAndGet() % 10^9)
```

lmk your preference